spec/lib/jobs/ipmi-sdr-alert-job-spec.js

```diff
@@ -131,12 +227,462 @@ describe(require('path').basename(__filename), function () {
+
+            var expectedReading = [
+                {
+                     sensorId: 'bad discrete sensor',
+                     entityId: '10.1 (Power Supply)',
+                     sdrType: 'Discrete',
+                     sensorType: 'Power Supply (0x08)',
+                     sensorReading: 'ffh',
+                     eventMessageControl: 'Per-threshold',
+                     assertionsEnabled: [
+                         'Presence detected',
+                         'Failure detected',
+                         'Power Supply AC lost'
+                     ],
+                     deassertionsEnabled: [
+                        'Presence detected',
+                        'Failure detected',
+                        'Power Supply AC lost'
+                     ],
+                     oem: '0',
+                     stateAsserted: 'Power Supply AC lost'
+                },
+                {
+                     sensorId: 'bad discrete sensor',
+                     entityId: '10.1 (Power Supply)',
+                     sdrType: 'Discrete',
+                     sensorType: 'Power Supply (0x08)',
+                     sensorReading: 'ffh',
+                     eventMessageControl: 'Per-threshold',
+                     assertionsEnabled: [
+                         'Presence detected',
+                         'Failure detected',
+                         'Power Supply AC lost'
+                     ],
+                     deassertionsEnabled: [
+                        'Presence detected',
+                        'Failure detected',
+                        'Power Supply AC lost'
+                     ],
+                     oem: '0',
+                     stateAsserted: 'Predictive failure'
+                },
+                {
+                     sensorId: 'bad discrete sensor',
+                     entityId: '10.1 (Power Supply)',
+                     sdrType: 'Discrete',
+                     sensorType: 'Power Supply (0x08)',
+                     sensorReading: 'ffh',
+                     eventMessageControl: 'Per-threshold',
+                     assertionsEnabled: [
+                         'Presence detected',
+                         'Failure detected',
+                         'Power Supply AC lost'
+                     ],
+                     deassertionsEnabled: [
+                        'Presence detected',
+                        'Failure detected',
+                        'Power Supply AC lost'
+                     ],
+                     oem: '0',
+                     stateAsserted: 'Failure detected'
+                },
+            ];
+            var expectedInCondition = [true, true, false];
+            badDiscreteTestSensor.statesAsserted.push('Predictive failure');
+            data.sdr = samples.concat(badDiscreteTestSensor);
+
+            return this.determineAlert(data).then(function(out) {
+                expect(out).to.have.length(3);
+                for (var i in out) {
```

Comment:
@benbp Sorry;  my comment wasn't clear.  The stackoverflow article was saying that it's bad practice to use a for/in loop to iterate over an array, not that forEach is bad.The only reason I suggested an alternative to the forEach loop is that I'm actually using elements from three different arrays inside the loop, so it seemed more convenient to have an index variable to use with all of them.